src/core/conversion/dataIndexer.ts

```diff
@@ -0,0 +1,157 @@
+        }
+        this.indices = Array.from({ length: Object.keys(completeMap).length }, () => 0);
+        this.lastIndices = Object.keys(targetMap).map(v => targetMap[v].length - 1);
+        this.lastCoordinateIndex = this.coordinates.length - 1;
+        this.reverseCumulativeProducts = this.generateRCP(variableSizes);
+        this.dataLength = Object.keys(targetMap).map(v => targetMap[v].length).reduce((acc, val) => acc * val, 1);
+        this.dataIndex = 0;
+        this.setCurrentIndex();
+    }
+
+    public getViewSeries(viewPrecision: number | null): IDataSeries[] {
+        const viewSeries: IDataSeries[] = [];
+        do {
+            const { rowNameGroup, dataCell } = this.calculateIndex(viewPrecision);
+            const group: IDataSeries | undefined = viewSeries.find(s => s.rowNameGroup.every(g => rowNameGroup.includes(g) && rowNameGroup.every(g => s.rowNameGroup.includes(g))));
+            if (group) {
+                group.series.push(dataCell);
+            }
+            else {
+                viewSeries.push({
+                    rowNameGroup: rowNameGroup,
+                    series: [dataCell]
+                })
+            }
+        } while (this.next());
+
+        return viewSeries;
+    }
+
+    next(): boolean {
+        for (let i = this.lastCoordinateIndex; i >= 0; i--) {
+            const variableIndex: number = this.variableOrder[i];
+            if (this.indices[variableIndex] < this.lastIndices[i]) {
+                this.indices[variableIndex]++;
+                this.setCurrentIndex();
+                return true;
+            }
+            else {
+                this.indices[variableIndex] = 0;
+            }
+        }
+        return false;
+    }
+
+    calculateIndex(viewPrecision: number | null): { rowNameGroup: TMultiLanguageString[], dataCell: IDataCell } {
+        let values: IVariableValueMeta[] = [];
+        let rowNameGroup: TMultiLanguageString[] = [];
+        for (let i = 0; i < this.indices.length; i++) {
+            const variable: IVariableMeta = this.responseObj.metaData[i];
+            const value: IVariableValueMeta = variable.values[this.indices[i]];
+            values.push(value);
+            if ((this.responseObj.rowVariableCodes.includes(variable.code) ||
+                this.responseObj.visualizationSettings.multiselectableVariableCode === variable.code) &&
+                variable.values.length > 1) {
+                rowNameGroup.push(value.name);
+            }
+        }
+        const dataCell: IDataCell = {
+            value: this.responseObj.data[this.dataIndex],
+            precision: viewPrecision ?? values.find(v => v.contentComponent)?.contentComponent?.numberOfDecimals ?? 0,
+            preliminary: values.some(v => Object.values(v.name)[0].trim().endsWith('*'))
+        };
+        if (!dataCell.value) dataCell.missingCode = this.responseObj.missingDataInfo[this.dataIndex];
+        const item = {
+            rowNameGroup: rowNameGroup,
```

Comment:
The building of the name groups should be separated from the cell building if possible.